src/service/matchers/lineage_matcher.py

```diff
@@ -3,36 +3,35 @@
 """
 
 import logging
+import re
+from typing import Any, Self, Set
 
-from pydantic import ConfigDict, BaseModel, Field
-from typing import Any
+from pydantic import ConfigDict, BaseModel, Field, model_validator, constr
 
 from src.common.constants import GTDB_UNCLASSIFIED_PREFIX
 from src.common.gtdb_lineage import GTDBRank, GTDBLineage, GTDBLineageError
 from src.common.storage.collection_and_field_names import FLD_GENOME_ATTRIBS_GTDB_LINEAGE
 from src.service import errors
 from src.service import models
 from src.service.app_state_data_structures import PickleableDependencies
-from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 from src.service.data_products import genome_attributes
 from src.service.matchers.common_models import Matcher
-
+from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 
 # See the workspace specs for the types listed in the matcher
 _GTDB_LINEAGE_METADATA_KEY = "GTDB_lineage"
 _GTDB_LINEAGE_VERSION_METADATA_KEY = "GTDB_source_ver"
 
-_COLLECTION_PARAMS_GTDB_VERSION_KEY = "gtdb_version"
+_COLLECTION_PARAMS_GTDB_VERSIONS_KEY = "gtdb_versions"
 
 
 class GTDBLineageMatcherCollectionParameters(BaseModel):
     "Parameters for the GTDB lineage matcher."
-    gtdb_version: str = Field(
-        example="207.0",
-        description="The GTDB version of the collection in which the matcher is installed. " +
-            "Input data to the matcher must match this version of GTDB or the match will " +
+    gtdb_versions: Set[constr(pattern=r"^\d{2,4}\.\d{1,2}$")] = Field(
+        example={"214.0", "214.1"},
```

Comment:
can we specify that the list has to be > 0 items long? I think theres a conlist function that'll do that

Reply:
👍

Reply:
Is it not possible to use a constr inside a conlist?